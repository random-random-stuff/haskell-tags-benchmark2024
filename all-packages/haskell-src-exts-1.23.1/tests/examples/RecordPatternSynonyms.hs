{-# LANGUAGE PatternSynonyms #-}
module ShouldCompile where

pattern Single{x} = [x]

pattern Double{y,z} = (y,z)

pattern More{x} <- (x,_) where
  More x = (x, Nothing)

-- Selector
selector :: Int
selector = x [5]

update :: [String]
update = ["String"] { x = "updated" }
