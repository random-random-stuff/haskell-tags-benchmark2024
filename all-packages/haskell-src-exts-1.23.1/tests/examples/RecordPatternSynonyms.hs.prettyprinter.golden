{-# LANGUAGE PatternSynonyms #-}
module ShouldCompile where

pattern Single{x} = [x]

pattern Double{y, z} = (y, z)

pattern More{x} <- (x, _)
  where More x = (x, Nothing)

selector :: Int
selector = x [5]

update :: [String]
update = ["String"]{x = "updated"}
